src/main.ts

```diff
@@ -88,15 +88,15 @@ export function validateObjectTree(iobObjects: ObjectWithValue[]): void {
 	// Foreach sub-state (. in id)
 	iobObjects.filter(item => item.object.type.split(".").length > 0).filter(item => item.object.type === "state").forEach(iobObj => {
 		const iobIDPath = iobObj.id.split(".");
-		if (iobIDPath.length > 0){
+		if (iobIDPath.length > 2){
 			// Remove last element (=name of state)
 			iobIDPath.pop();
 			let iobIDBasePath = "";
 			iobIDPath.forEach(iobIDName =>{
 				if (iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length > 1){
 					throw `Duplicated object ${iobIDBasePath}${iobIDName} defined`
 				}else{
-					if (iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length === 2){
+					if (iobIDPath.indexOf(iobIDName) >= 2 && iobObjects.filter(item => item.id === `${iobIDBasePath}${iobIDName}`).length === 0){
```

Comment:
We'll probably need to set up some automatic testing for this method. This looks better than before but what happens if I have such an id: `adapter.0.foobar.foobar.foobar`? Can this method handle that the same id part exists multiple times?
indexOf only finds the first one.